inquire/src/prompts/path_select/mod.rs

```diff
@@ -0,0 +1,365 @@
+            .collect::<Vec<String>>()
+            .join(", ")
+    };
+
+    /// Default filter function, which only checks if the END component (file name, directory name)
+    /// of the path is a match for the filter
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use inquire::PathSelect;
+    ///
+    /// let filter = PathSelect::<&str>::DEFAULT_FILTER;
+    /// assert_eq!(false, filter("fer", &"/nefer/osiris/hotep/ptah/maat",  "/nefer/osiris/hotep/ptah/maat",  0));
+    /// assert_eq!(false, filter("tep", &"/nefer/osiris/hotep/ptah/maat",  "/nefer/osiris/hotep/ptah/maat",  1));
+    /// assert_eq!(true, filter("aa", &"/nefer/osiris/hotep/ptah/maat",  "/nefer/osiris/hotep/ptah/maat",  2));
+    /// assert_eq!(true, filter("ma", &"/nefer/osiris/hotep/ptah/maat",  "/nefer/osiris/hotep/ptah/maat",  3));
+    /// assert_eq!(true, filter("ma", &"/nefer/osiris/hotep/ptah/Maat",  "/nefer/osiris/hotep/ptah/Maat",  4));
+    /// assert_eq!(true, filter("ma", &"/nefer/osiris/hotep/ptah/Maat.rs",  "/nefer/osiris/hotep/ptah/Maat.rs",  5));
+    /// assert_eq!(true, filter("Ma", &"/nefer/osiris/hotep/ptah/maat.rs",  "/nefer/osiris/hotep/ptah/maat.rs",  5));
+    /// ```
+    pub const DEFAULT_FILTER: Filter<'a, T> = &|filter, as_ref_path, _, _| -> bool {
+        let filter = filter.to_lowercase();
+        as_ref_path
+            .as_ref()
+            .file_name()
+            .unwrap_or_default()
+            .to_string_lossy()
+            .to_lowercase()
+            .contains(&filter)
+    };
+
+    /// Default page size, equal to the global default page size [config::DEFAULT_PAGE_SIZE]
+    pub const DEFAULT_PAGE_SIZE: usize = crate::config::DEFAULT_PAGE_SIZE;
+
+    /// Default value of vim mode, equal to the global default value [config::DEFAULT_PAGE_SIZE]
+    pub const DEFAULT_VIM_MODE: bool = crate::config::DEFAULT_VIM_MODE;
+
+    /// Default value of showing hidden files
+    pub const DEFAULT_SHOW_HIDDEN: bool = false;
+
+    /// Default help message.
+    pub const DEFAULT_HELP_MESSAGE: Option<&'a str> = Some(
+        "↑↓ to move, space to select one, \
+        → to navigate to path, ← to navigate up, \
+        shift+→ to select all, shift+← to clear, \
+        tab to change sorting mode",
+    );
+
+    /// Default behavior of keeping or cleaning the current filter value.
+    pub const DEFAULT_KEEP_FILTER: bool = true;
+
+    /// Default value of showing symlinks
+    pub const DEFAULT_SHOW_SYMLINKS: bool = false;
+
+    /// Default value of selecting multiple files
+    pub const DEFAULT_SELECT_MULTIPLE: bool = false;
+
+    /// Default visual divider value.
+    pub const DEFAULT_DIVIDER: &'a str = "-----";
+
+    /// Creates a [PathSelect] with the provided message and options, along with default configuration values.
+    pub fn new(message: &'a str, start_path_opt: Option<T>) -> Self {
+        Self {
+            message,
+            start_path_opt,
```

Comment:
If it's optional, let's make users set it via the builder and allow the happy path to be frictionless

```suggestion
    pub fn new(message: &'a str) -> Self {
        Self {
            message,
            start_path_opt: None,
```